library/ssl_tls.c

```diff
@@ -1984,6 +2003,19 @@ static int ssl_tls13_session_save( const mbedtls_ssl_session *session,
     memcpy( p, session->resumption_key, session->resumption_key_len );
     p += session->resumption_key_len;
 
+#if defined(MBEDTLS_SSL_SERVER_NAME_INDICATION) && defined(MBEDTLS_SSL_CLI_C)
+    if( session->endpoint == MBEDTLS_SSL_IS_CLIENT )
+    {
+        p[0] = session->hostname_len;
+        p++;
```

Comment:
align with `hostname` definition
```suggestion
        MBEDTLS_PUT_UINT16_BE( session->hostname_len p, 0 );
        p += 2;
```